Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,261 @@
+            string[][] workingPeopleAndBalances = ArrayifyBalances(peopleAndBalances);
+            float biggestLossEver = GetLoss(workingPeopleAndBalances[0]);
+            string biggestLossNames = workingPeopleAndBalances[0][0];
+            for (int i = 0; i < workingPeopleAndBalances.Length; i++)
+            {
+                float loss = GetLoss(workingPeopleAndBalances[i]);
+                string name = workingPeopleAndBalances[i][0];
+                if (loss < biggestLossEver)
+                {
+                    biggestLossEver = loss;
+                    biggestLossNames = name;
+                }
+            }
+
+            return $"{biggestLossNames} lost the most money. {currencySymbol}{biggestLossEver}";
         }
 
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return invalidMessage;
+            }
+
+            string[][] workingBalances = ArrayifyBalances(peopleAndBalances);
+            int richestPerson = 0;
+            string message = $"{workingBalances[richestPerson][0]} is the richest person. ${workingBalances[richestPerson][^1]}";
+            if (workingBalances.Length == 1)
+            {
+                return message;
+            }
+            for (int i = 1; i < workingBalances.Length; i++)
+            {
+                if(float.Parse(workingBalances[i][^1]) > float.Parse(workingBalances[richestPerson][^1]))
+                {
+                    richestPerson = i;
+                }
+            }
+            return message = $"{workingBalances[richestPerson][0]} is the richest person. ${workingBalances[richestPerson][^1]}";
         }
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return invalidMessage;
+            }
+
+            string[][] workingBalances = ArrayifyBalances(peopleAndBalances);
+            int poorestPerson = 0;
+            string message = $"{workingBalances[poorestPerson][0]} is the poorest person. ${workingBalances[poorestPerson][^1]}";
+            if (workingBalances.Length == 1)
+            {
+                return message;
+            }
+            for (int i = 1; i < workingBalances.Length; i++)
+            {
+                if (float.Parse(workingBalances[i][^1]) < float.Parse(workingBalances[poorestPerson][^1]))
+                {
+                    poorestPerson = i;
+                }
+            }
+            return message = $"{workingBalances[poorestPerson][0]} is the poorest person. ${workingBalances[poorestPerson][^1]}";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'message'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)